pkg/provider/gke/gke.go

```diff
@@ -79,6 +79,20 @@ func (c *GKE) NewGKEClient(*kingpin.ParseContext) error {
 		os.Setenv("GOOGLE_APPLICATION_CREDENTIALS", c.Auth)
 		c.Auth = string(content)
 	} else {
+		// Check if auth data is base64 encoded and decode it.
```

Comment:
For complitness I think it is better to check and decode in both cases, when it is a file and when it is a data string in an env var.

Reply:
In that case, if we find that the file is base64  encoded, we'll need to override the file with the decoded data because `GOOGLE_APPLICATION_CREDENTIALS` expects the path of the file, content of the file should not be base64 encoded.

Reply:
or we can:
1. check if it's file or data
2. store the content of the file in `Auth` if it's a file
3. check if `Auth` is encoded, decode it,
4. create a temporary file to store the decoded data
5. `os.SetEnv(GOOGLE_APPLICATION_CREDENTIALS, toTheTempFile)`

what do  you think? this way create a temporary file no matter what and there's only one call to set `GOOGLE_APPLICATION_CREDENTIALS`

Reply:
yes that sound good.

Reply:
updated.